### DISCUSSION TITLE
Diffusion python images

### DISCUSSION THREAD

**Discussion by christopher5106** [huggingface/diffusers#9176] [General]
I'm wondering about this normalization during training
https://github.com/huggingface/diffusers/blob/main/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1079
that means images are between 0 and 1
while image preprocessing and post processing deal with pytorch images between -1 and 1 usually ?
do you have any insight about that ?

**Comment by tolgacangoz** [comment]
Could you examine [`torchvision.transforms.ToTensor()`](https://pytorch.org/vision/stable/generated/torchvision.transforms.ToTensor.html) and [`torchvision.transforms.Normalize()`](https://pytorch.org/vision/main/generated/torchvision.transforms.Normalize.html); you will see how they convert the data.

**Reply by tolgacangoz** [reply]
`ToTensor()` makes them [0, 1], and `Normalize()` makes them [-1, 1]. Could you elaborate on the meaning of this sentence: "that means images are between 0 and 1"?

**Comment by christopher5106** [comment]
I know, but my question is more about the usage of normalisation to -1:1  in diffusion:

https://github.com/huggingface/diffusers/blob/main/src/diffusers/image_processor.py#L152

**Reply by asomoza** [reply]
I haven't tested this and I'm not the one that implemented the code, but I can think of two reasons:

- The [original implementation](https://github.com/runwayml/stable-diffusion/blob/08ab4d326c96854026c4eb3454cd3b02109ee982/scripts/img2img.py#L57) and training used this normalization
- Related to one also, neural networks often perform better when working with input values centered around zero.

[Here](https://stats.stackexchange.com/a/422087) is a not too long more detailed explanation of the last point that I liked.

**Comment by christopher5106** [comment]
Ah sorry I missed the point that normalise bring images to range -1,1

Thank you for your help

**Reply by asomoza** [reply]
oh so your question was just that you didn't understood the difference between vanilla pytorch and torchvision. I also misunderstood your question.
